server/services/user.js

```diff
@@ -90,6 +90,19 @@ function removeGroup(user, groupName) {
     .then(validatedName => user.removeGroup(validatedName));
 }
 
+function getGroupsForUser(user) {
+  return new Promise((resolve, reject) => {
+    User.findOne(user)
+    .populate('activeGroups')
+    .exec((err, u) => {
```

Comment:
The initial comment got removed from here, but the `.findOne().exec()` will return a promise that you can work with